include/bluetooth/mesh/main.h

```diff
@@ -105,6 +105,23 @@ struct bt_mesh_prov {
 	/** Out of Band information field. */
 	bt_mesh_prov_oob_info_t oob_info;
 
+	/** Pointer to Public Key in big-endian for OOB public key type support.
+	 *
+	 * Remember to enable :c:option:`CONFIG_BT_MESH_PROV_OOB_PUBLIC_KEY`
+	 * when initializing this parameter.
+	 *
+	 * Must be used together with :c:member:`struct bt_mesh_prov.private_key_be.
+	 */
+	const uint8_t *public_key_be;
+	/** Pointer to Private Key in big-endian for OOB public key type support.
+	 *
+	 * Remember to enable :c:option:`CONFIG_BT_MESH_PROV_OOB_PUBLIC_KEY`
+	 * when initializing this parameter.
+	 *
+	 * Must be used together with :c:member:`struct bt_mesh_prov.public_key_be.`
```

Comment:
Perhaps we could return an error from `bt_mesh_prov_enable` if only one is non-null? E.g.
```c
	if (!!bt_mesh_prov->public_key_be != !!bt_mesh_prov->private_key_be) {
		BT_WARN("Both public and private key must be set");
		return -EINVAL;
	}
```

Reply:
There is the warning here: https://github.com/zephyrproject-rtos/zephyr/blob/04e81153e066577d39a96a209d9444182cd03aa3/subsys/bluetooth/mesh/prov_device.c#L145